paimon-core/src/main/java/org/apache/paimon/table/source/MergeTreeSplitGenerator.java

```diff
@@ -61,11 +61,17 @@ public MergeTreeSplitGenerator(
     }
 
     @Override
-    public List<List<DataFileMeta>> splitForBatch(List<DataFileMeta> files) {
-        if (deletionVectorsEnabled || mergeEngine == FIRST_ROW) {
+    public List<SplitGroup> splitForBatch(List<DataFileMeta> files) {
+        boolean rawConvertible = files.stream().allMatch(DataFileMeta::rawConvertible);
+        boolean oneLevel =
```

Comment:
oneLevel and not zero level.

You should add test for this.

Reply:
move rawConvertible to avoid potential problems with the append table